### PR TITLE
devicetree: use new macros for zephyr,dtcm & zephyr,ccm

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: include/arch/arm/aarch32/cortex_m/scripts/linker.ld [hunk 1]
```diff
@@ -98,8 +98,9 @@ MEMORY
 #ifdef DT_CCM_BASE_ADDRESS
     CCM                   (rw) : ORIGIN = DT_CCM_BASE_ADDRESS, LENGTH = DT_CCM_SIZE * 1K
 #endif
-#ifdef DT_DTCM_BASE_ADDRESS
-    DTCM                  (rw) : ORIGIN = DT_DTCM_BASE_ADDRESS, LENGTH = DT_DTCM_SIZE * 1K
+#if DT_HAS_NODE(DT_CHOSEN(zephyr_dtcm))
+#warning FOOOO
```
Comment (erwango) [inline include/arch/arm/aarch32/cortex_m/scripts/linker.ld:102]: Leftover ?

**Review by erwango — CHANGES_REQUESTED** [review]
Dev leftover I guess + checkpatch to be fixed

File: include/arch/arm/aarch32/cortex_m/scripts/linker.ld (hunk 1 above)
Comment (galak) [reply include/arch/arm/aarch32/cortex_m/scripts/linker.ld:102]: Yep

Comment (galak) [reply include/arch/arm/aarch32/cortex_m/scripts/linker.ld]: fixed.

**Review by erwango — APPROVED** [review]
Fine w/ the change other than checkpatch that might have to be fixed

**Comment by galak** [comment]
> Fine w/ the change other than checkpatch that might have to be fixed

fixed

**Pull request merged**